pinot-server/src/main/java/org/apache/pinot/server/starter/helix/SegmentMessageHandlerFactory.java

```diff
@@ -173,6 +177,17 @@ public HelixTaskResult handleMessage()
       try {
         _instanceDataManager.deleteTable(_tableNameWithType);
         helixTaskResult.setSuccess(true);
+        Arrays.stream(ServerMeter.values())
+            .filter(m -> !m.isGlobal())
+            .forEach(m -> _metrics.removeTableMeter(_tableNameWithType, m));
+        Arrays.stream(ServerGauge.values())
+            .filter(g -> !g.isGlobal())
+            .forEach(g -> _metrics.removeTableGauge(_tableNameWithType, g));
+        Arrays.stream(ServerTimer.values())
+            .filter(t -> !t.isGlobal())
+            .forEach(t -> _metrics.removeTableTimer(_tableNameWithType, t));
+        Arrays.stream(ServerQueryPhase.values())
+            .forEach(p -> _metrics.removePhaseTiming(_tableNameWithType, p));
       } catch (Exception e) {
         _metrics.addMeteredTableValue(_tableNameWithType, ServerMeter.DELETE_TABLE_FAILURES, 1);
```

Comment:
this `ServerMeter.DELETE_TABLE_FAILURES` is also a table level metrics right?

Reply:
Yes, it is. A sequence like:
1. Create table A
2. Fail to delete table A
3. Successfully delete table A

Will show DELETE_TABLE_FAILURES with value 1 for table A point 2 and 3, but won't show that value after 3 is correctly executed.

Reply:
Will the next run of `SegmentStatusChecker` clear this metric accidentally? I guess we want to keep it, or change it to a global metric?

Reply:
I'm going to remove this code because that is not correctly deleting the server metrics. It seems even after `        _instanceDataManager.deleteTable(_tableNameWithType);` is called other threads are updating these metrics, so some of the deleted metrics here are created again. Specifically the ones that survive are:
```
"org.apache.pinot.common.metrics":name="pinot.server.endToEndRealtimeIngestionDelayMs.meetupRsvp_REALTIME.X",type="ServerMetrics"
"org.apache.pinot.common.metrics":name="pinot.server.highestStreamOffsetConsumed.meetupRsvp_REALTIME-meetupRSVPEvents-X",type="ServerMetrics"
"org.apache.pinot.common.metrics":name="pinot.server.lastRealtimeSegmentCompletionDurationSeconds.meetupRsvp_REALTIME-meetupRSVPEvents-X",type="ServerMetrics"
"org.apache.pinot.common.metrics":name="pinot.server.lastRealtimeSegmentCreationDurationSeconds.meetupRsvp_REALTIME-meetupRSVPEvents-X",type="ServerMetrics"
"org.apache.pinot.common.metrics":name="pinot.server.lastRealtimeSegmentInitialConsumptionDurationSeconds.meetupRsvp_REALTIME-meetupRSVPEvents-X",type="ServerMetrics"
"org.apache.pinot.common.metrics":name="pinot.server.meetupRsvp.segmentUploadSuccess",type="ServerMetrics"
"org.apache.pinot.common.metrics":name="pinot.server.meetupRsvp.segmentUploadTimeMs",type="ServerMetrics"
"org.apache.pinot.common.metrics":name="pinot.server.meetupRsvp_REALTIME-meetupRSVPEvents-X.realtimeRowsConsumed",type="ServerMetrics"
"org.apache.pinot.common.metrics":name="pinot.server.realtimeOffheapMemoryUsed.meetupRsvp",type="ServerMetrics"

```

Reply:
Nope, `SegmentStatusChecker` cleans the controller metrics. This is a server metric.

Anyway I've changed the code a bit. Semantic is almost the same, but it will be cleaner that `ServerMeter.DELETE_TABLE_FAILURES` will not be deleted in case there is an exception.